cmd/jujud/agent/caasoperator.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package agent
+
+import (
+	"runtime"
+	"time"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/loggo"
+	"github.com/juju/utils/featureflag"
+	"github.com/juju/utils/voyeur"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/juju/names.v2"
+	worker "gopkg.in/juju/worker.v1"
+	"gopkg.in/natefinch/lumberjack.v2"
+	"gopkg.in/tomb.v1"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api/base"
+	//	apicaasoperator "github.com/juju/juju/api/caasoperator"
+	"github.com/juju/juju/cmd/jujud/agent/caasoperator"
+	cmdutil "github.com/juju/juju/cmd/jujud/util"
+	jujuversion "github.com/juju/juju/version"
+	jworker "github.com/juju/juju/worker"
+	"github.com/juju/juju/worker/dependency"
+	"github.com/juju/juju/worker/introspection"
+	"github.com/juju/juju/worker/logsender"
+)
+
+var (
+
+	// should be an explicit dependency, can't do it cleanly yet
+	caasOperatorManifolds = caasoperator.Manifolds
+)
+
+// CaasOperatorAgent is a cmd.Command responsible for running a CAAS operator agent.
+type CaasOperatorAgent struct {
+	cmd.CommandBase
+	tomb tomb.Tomb
+	AgentConf
+	configChangedVal *voyeur.Value
+	CaasOperatorName string
+	runner           *worker.Runner
+	bufferedLogger   *logsender.BufferedLogWriter
+	setupLogging     func(agent.Config) error
+	logToStdErr      bool
+	ctx              *cmd.Context
+
+	// Used to signal that the upgrade worker will not
+	// reboot the agent on startup because there are no
+	// longer any immediately pending agent upgrades.
+	// Channel used as a selectable bool (closed means true).
+	initialUpgradeCheckComplete chan struct{}
+
+	prometheusRegistry *prometheus.Registry
+}
+
+func NewCaasOperatorAgent(ctx *cmd.Context, bufferedLogger *logsender.BufferedLogWriter) (*CaasOperatorAgent, error) {
+	prometheusRegistry, err := newPrometheusRegistry()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	// MMCC debug
+	rootlogger := loggo.GetLogger("")
+	rootlogger.SetLogLevel(loggo.TRACE)
+	logger.Infof("In NewCaasOperatorAgent")
+	// end
+	return &CaasOperatorAgent{
+		AgentConf:        NewAgentConf(""),
+		configChangedVal: voyeur.NewValue(true),
+		ctx:              ctx,
+		initialUpgradeCheckComplete: make(chan struct{}),
+		bufferedLogger:              bufferedLogger,
+		prometheusRegistry:          prometheusRegistry,
+	}, nil
+}
+
+func (op *CaasOperatorAgent) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "caasoperator",
+		Purpose: "run a juju CAAS Operator",
+	}
+}
+
+func (op *CaasOperatorAgent) SetFlags(f *gnuflag.FlagSet) {
+	op.AgentConf.AddFlags(f)
+	f.StringVar(&op.CaasOperatorName, "caasoperator-name", "", "name of the operator to run")
```

Comment:
I think this would make more sense as "--application-name". The existing machine agent takes the machine and the unit agent takes the unit name.